include/envoy/network/filter.h

```diff
@@ -232,6 +234,47 @@ class ListenerFilterManager {
  */
 typedef std::function<void(ListenerFilterManager& filter_manager)> ListenerFilterFactoryCb;
 
+/**
+ * Interface representing a single filter chain.
+ */
+class FilterChain {
+public:
+  virtual ~FilterChain() {}
+
+  /**
```

Comment:
To be honest, I don't see much value in exposing implementation details / member classes, since it makes it harder to make changes in the future.

What would be benefit of returning `TransportSocketFactory` directly here? Calls to those 2 methods are done in separate places, so it wouldn't save any calls.

Reply:
It is not exposing implementation details / member classes, the `TransportSocketFactory` is a pure interface that could be returned here. Just to be more consistent with cluster, also avoid copy and paste same interface around.